axelor-account/src/main/java/com/axelor/apps/account/service/analytic/AnalyticMoveLineQueryServiceImpl.java

```diff
@@ -108,82 +106,55 @@ public String getAnalyticMoveLineQuery(AnalyticMoveLineQuery analyticMoveLineQue
+                    parameter.getAnalyticAxis().getId(),
+                    parameter.getAnalyticAccountSet().stream()
+                        .map(AnalyticAccount::getId)
+                        .map(String::valueOf)
+                        .collect(Collectors.joining(","))))
+        .collect(Collectors.joining(" OR "));
   }
 
   protected String getAndQuery(
       String query, List<AnalyticMoveLineQueryParameter> searchAnalyticMoveLineQueryParameterList) {
-    Map<MoveLine, List<AnalyticMoveLine>> analyticMoveLineList =
-        Beans.get(AnalyticMoveLineRepository.class)
-            .all()
-            .filter("self.moveLine is not null")
-            .fetch()
-            .stream()
-            .collect(Collectors.groupingBy(AnalyticMoveLine::getMoveLine, Collectors.toList()));
-
     Map<AnalyticAxis, Set<AnalyticAccount>> paramMap =
         searchAnalyticMoveLineQueryParameterList.stream()
             .collect(
                 Collectors.toMap(
                     AnalyticMoveLineQueryParameter::getAnalyticAxis,
                     AnalyticMoveLineQueryParameter::getAnalyticAccountSet));
 
-    List<Long> filteredList = new ArrayList<>();
-    filteredList.add(0l);
-
-    for (Map.Entry<MoveLine, List<AnalyticMoveLine>> entry : analyticMoveLineList.entrySet()) {
-      Map<AnalyticAxis, List<AnalyticMoveLine>> val =
-          entry.getValue().stream()
-              .collect(Collectors.groupingBy(AnalyticMoveLine::getAnalyticAxis));
-      Boolean allCondition = true;
-      for (Map.Entry<AnalyticAxis, List<AnalyticMoveLine>> entry2 : val.entrySet()) {
-
-        Set<AnalyticAccount> analyticAccountSet = paramMap.get(entry2.getKey());
-
-        if (analyticAccountSet == null) {
-          continue;
-        }
-
-        if (entry2.getValue().stream()
-            .noneMatch(a -> analyticAccountSet.contains(a.getAnalyticAccount()))) {
-          allCondition = false;
-          break;
-        }
-      }
-      if (allCondition) {
-        filteredList.add(entry.getKey().getId());
-      }
-    }
+    String filteredIds =
+        analyticMoveLineRepository.all().filter("self.moveLine is not null").fetch().stream()
+            .collect(Collectors.groupingBy(AnalyticMoveLine::getMoveLine))
+            .entrySet()
+            .stream()
+            .filter(
+                entry ->
+                    paramMap.entrySet().stream()
+                        .allMatch(
+                            param ->
+                                entry.getValue().stream()
+                                    .anyMatch(
+                                        aml ->
+                                            aml.getAnalyticAxis().equals(param.getKey())
+                                                && param
+                                                    .getValue()
+                                                    .contains(aml.getAnalyticAccount()))))
```

Comment:
please improve readability for this code, this is too nested.

Also check if you can move checks from the java filter to the JPQL filter, it will improve performance.